includes/admin/class-wc-amazon-payments-advanced-order-admin.php

```diff
@@ -132,7 +132,11 @@ public function meta_box() {
 			return;
 		}
 
-		add_meta_box( 'woocommerce-amazon-payments-advanced', __( 'Amazon Pay', 'woocommerce-gateway-amazon-payments-advanced' ), array( $this, 'authorization_box' ), 'shop_order', 'side' );
+		$post_types = apply_filters( 'woocommerce_amazon_pa_admin_meta_box_post_types', array( 'shop_order' ) );
+
+		foreach ( $post_types as $post_type ) {
```

Comment:
You may add here a check for the subscription parent orders with  0 as total, to remove the actions.